metric_source/prometheus/fetch.go

```diff
@@ -15,7 +15,28 @@ import (
 func (prometheus *Prometheus) Fetch(target string, from, until int64, allowRealTimeAlerting bool) (metricSource.FetchResult, error) {
 	from = moira.MaxInt64(from, until-int64(prometheus.config.MetricsTTL.Seconds()))
 
-	ctx, cancel := context.WithTimeout(context.Background(), prometheus.config.Timeout)
+	var err error
+	for i := 0; i < prometheus.config.Retries; i++ {
+		var res metricSource.FetchResult
+		res, err = prometheus.fetch(target, from, until, allowRealTimeAlerting)
+
+		if err == nil {
+			return res, nil
+		}
+
+		prometheus.logger.Warning().
+			Error(err).
+			Int("retries left", prometheus.config.Retries-i-1).
+			Msg("Failed to fetch prometheus target")
+
+		time.Sleep(prometheus.config.RetryTimeout)
```

Comment:
why should we sleep if it is last available attempt?

Reply:
Fixed